### PR TITLE
Simplify setup.py and require Cython.

### PR DESCRIPTION
**Pull request by socketpair**

### CODE REVIEW

**Comment by socketpair** [comment]
1. Changes according to https://setuptools.readthedocs.io/en/latest/setuptools.html#distributing-extensions-compiled-with-cython
2. I insist on not distributing .c-files in git-repo at least. All systems having Python3 also have Cython definitely. If you know exact issues with some version of Cython -- just bump it in the requirements.

File: setup.py [hunk 1]
```diff
@@ -1,71 +1,34 @@
...
-from setuptools.extension import Extension
+from setuptools import Extension, setup, find_packages
 
 __module__ = 'adbus'
 __url__ = 'https://github.com/ccxtechnologies'
 
 __version__ = None
 exec(open(f'{__module__}/__version__.py').read())
 
-if "--nosystemd" in sys.argv:
```
Comment (socketpair) [inline setup.py:16]: I can't understand that, seems some non-reachable code.

File: setup.py [hunk 2]
```diff
@@ -1,71 +1,34 @@
...
-        ]
-
-
-def check_external_dependancy(name):
-    try:
-        whereis = subprocess.run(
-                ['whereis', name], check=True, stdout=subprocess.PIPE
-        )
-        if b'/' not in whereis.stdout:
-            print(f"\033[91m Failed to find dependency {name}.\033[0m")
```
Comment (socketpair) [inline setup.py:48]: Writing escape-sequences to non-terminal is bogus.

File: setup.py [hunk 3]
```diff
@@ -1,71 +1,34 @@
...
-    else:
-        return [
-                Extension(
-                        f"{__module__}.{module}", [f"{__module__}/{module}.c"],
-                        libraries=libraries
-                )
-        ]
-
-
-def check_external_dependancy(name):
```
Comment (socketpair) [inline setup.py:42]: You should not check dependencies in that way. There are package / environment maintainers for that. For example, you don't check library version...Also you check for systemd, but not for libsystemd development files actually.

**Comment by charleseidsness** [comment]
We require the c file to be included in git, and we use those features as part of our build system.

Neither should have an impact on someone else using this library.

**Comment by socketpair** [comment]
OK, Will you merge PR if I change it to preserve .c file and not fail if Cython not installed ?

**Comment by charleseidsness** [comment]
Though I appreciate your interest in this library, and your enthusiasm, your updates primarily remove features that we've included to build the library using our build system. These features to my knowledge do not make the library un-buildable by others.

If there is a specific issue that you had building the library, and you have a fix to that issue that doesn't negatively effect our use of the library we'd be happy to review it.

This library is a part of a much larger production code-base. We provided it as open source software so that others may get some use out of it but we can't make any updates to it that will negatively impact the way that we use the library.

**Comment by socketpair** [comment]
Well, I changed PR. Now it works with Cython and without. Also (I hope) it will not break your build system. Please review.

**Comment by socketpair** [comment]
Well, I have updated the branch, but can not re-open PR. Please make new one if you are interested in.

**Comment by charleseidsness** [comment]
Thanks, but that adds a dependency on cython and breaks the way we build it. If you would rather build it that way that's great, but you can use your own fork for that.

**Comment by socketpair** [comment]
Well. I can fix even this. I just don't want to maintain a fork...We just start using you library for own purposes (and seems everything works), be sure, if we find bugs we will contribute fixes.

**Comment by charleseidsness** [comment]
That's great, I'm happy to hear that someone else may get some use out of the library.

If you find bugs, or have issues building we are definitely open to fixes. We obliviously can't make updates that will break anything that we are using, and typically we won't make updates that only change some minor style thing, like changing gitignore. Especially when we prefer to have a single gitignore file (just personal preference).

**Comment by jameshilliard** [comment]
> that adds a dependency on cython and breaks the way we build it

Do you build adbus as a buildroot package(I'm using it with buildroot myself)? I think buildroot recently added a host-cython package which should allow for building cython extensions directly.

**Comment by charleseidsness** [comment]
We do use buildroot. Interesting! We'll definitely take a look when we get a chance.

**Comment by jameshilliard** [comment]
I got this working with buildroot in #43.

**Pull request closed without merging**